tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -1526,23 +1537,68 @@ pub(super) mod test {
             .with_timer(MockTime);
         let expected = "fake time tracing_subscriber::fmt::format::test: hello\n";
 
-        run_test(subscriber, make_writer, expected);
+        assert_info_hello(subscriber, make_writer, expected);
     }
 
-    #[cfg(feature = "ansi")]
-    fn test_ansi(is_ansi: bool, expected: &str) {
+    #[test]
+    fn with_thread_ids() {
         let make_writer = MockMakeWriter::default();
         let subscriber = crate::fmt::Collector::builder()
             .with_writer(make_writer.clone())
-            .with_ansi(is_ansi)
+            .with_thread_ids(true)
+            .with_ansi(false)
+            .with_timer(MockTime);
+        let expected = "fake time  INFO ThreadId() tracing_subscriber::fmt::format::test: hello\n";
+
+        assert_info_hello_ignore_numeric(subscriber, make_writer, expected);
+    }
+
+    #[test]
+    fn pretty_default() {
+        let make_writer = MockMakeWriter::default();
+        let subscriber = crate::fmt::Collector::builder()
+            .pretty()
+            .with_writer(make_writer.clone())
+            .with_ansi(false)
             .with_timer(MockTime);
-        run_test(subscriber, make_writer, expected)
+        let expected = format!(
+            r#"  fake time  INFO tracing_subscriber::fmt::format::test: hello
+    at {}:
+
+"#,
+            PathBuf::from("tracing-subscriber/src/fmt/format/mod.rs")
+                .to_str()
+                .unwrap()
+        );
+
+        assert_info_hello_ignore_numeric(subscriber, make_writer, &expected)
     }
 
-    fn run_test(subscriber: impl Into<Dispatch>, buf: MockMakeWriter, expected: &str) {
+    fn assert_info_hello(subscriber: impl Into<Dispatch>, buf: MockMakeWriter, expected: &str) {
         let _default = set_default(&subscriber.into());
         tracing::info!("hello");
-        assert_eq!(expected, buf.get_string())
+        let result = buf.get_string();
+
+        assert_eq!(expected, result)
+    }
+
+    // When numeric characters are used they often form a non-deterministic value.
+    // This assert method should be used when non-deterministic numeric characters are present.
+    fn assert_info_hello_ignore_numeric(
+        subscriber: impl Into<Dispatch>,
+        buf: MockMakeWriter,
+        expected: &str,
+    ) {
+        let _default = set_default(&subscriber.into());
+        tracing::info!("hello");
+
+        let result: String = buf
+            .get_string()
+            .chars()
+            .filter(|x| !x.is_numeric())
+            .collect();
```

Comment:
i think this could probably be simplified a bit using [`str::replace`](https://doc.rust-lang.org/stable/std/string/struct.String.html#method.replace) from the standard library? not a hard blocker, though.

Reply:
also, it could be nice if we replaced the numeric characters with a specific character, rather than just removing them? that way, we would be asserting that they were present at that position in the expected string. the current tests will pass even if, for example, no line number was actually printed.

on the other hand, there might be a variable number of numeric characters in the string --- if the code changes significantly, the line number might have another tens place, for example. we could probably work around that using a regex to replace any number of numeric characters with a single one, but i am not sure if this is really worth the effort...

Reply:
I avoided solutions using regex to avoid adding it as a dev-dep. As you've noted we could make this a lot nicer with regex, so if your happy for me to add the regex dev-dep then I'll rewrite this using regex.